### PR TITLE
Issue #PS-2002 feat: workspace UI integration

### PR DESCRIPTION
**Pull request by suvarnakale**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    "module": "esnext",
+    "esModuleInterop": true,
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "jsx": "preserve"
+  },
+  "include": [
+    "next-env.d.ts",
+    "**/*.ts",
```
Comment (itsvick) [inline tsconfig.json:22]: @suvarnakale are we using jsconfing or tsconfig?

Comment (suvarnakale) [reply tsconfig.json:22]: I think we should use tsconfig bcoz we r going to integrate it with admin which also uses similar configurations so later conflicts will not arise.

File: src/components/CourseCard.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        flexDirection: "column",
+        width: "250px",
+      }}
+    >
+      <Box position="relative">
+        <CardMedia
+          component="div"
+          sx={{
+            height: 140,
+            backgroundColor: "#e0e0e0",
```
Comment (itsvick) [inline src/components/CourseCard.tsx:46]: @suvarnakale avoid hard coding color

File: src/components/CourseCard.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            display: "flex",
+            alignItems: "center",
+            justifyContent: "center",
+          }}
+        >
+          {imageUrl ? (
+            <img
+              src={imageUrl}
+              alt={title || "Untitled Course"}
```
Comment (itsvick) [inline src/components/CourseCard.tsx:55]: @suvarnakale remove or condition

File: src/components/CourseCard.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+              style={{ width: "100%", height: "100%", objectFit: "cover" }}
+            />
+          ) : (
+            <ImageIcon fontSize="large" />
+          )}
+        </CardMedia>
+
+        <Chip
+          label={type || "Course"}
```
Comment (itsvick) [inline src/components/CourseCard.tsx:64]: @suvarnakale remove or condition

File: src/components/CourseCard.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+          sx={{
+            position: "absolute",
+            top: 10,
+            right: 10,
+            backgroundColor: "black",
+            color: "white",
```
Comment (itsvick) [inline src/components/CourseCard.tsx:70]: @suvarnakale take colors from the theme file

File: src/components/CourseCard.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+          }}
+        />
+      </Box>
+
+      <CardContent sx={{ flex: 1 }}>
+        <Typography variant="h6">{title || "Untitled Course"}</Typography>
```
Comment (itsvick) [inline src/components/CourseCard.tsx:76]: @suvarnakale remove or condition from everywhere

File: src/components/CourseCard.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        <Typography variant="body2" color="text.secondary">
+          {description || "Enter description of course"}
+        </Typography>
+      </CardContent>
+
+      {(status === "draft" || status === "published") && (
```
Comment (itsvick) [inline src/components/CourseCard.tsx:82]: @suvarnakale create status as enum in app.constant file

File: src/components/SearchBox.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        }}
+      />
+
+      {/* Filter Dropdown */}
+      <FormControl sx={{ minWidth: 150 }}>
+        <InputLabel>Filter</InputLabel>
+        <Select value={filter} onChange={handleFilterChange} label="Filter">
+          {/* Add dummy filters here */}
+          <MenuItem value="all">All</MenuItem>
+          <MenuItem value="courses">Courses</MenuItem>
```
Comment (itsvick) [inline src/components/SearchBox.tsx:77]: @suvarnakale create a array of objects for label and value and use for loop

File: src/components/SideBar.tsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+            onClick={() => handleNavigation(item.key)}
+          >
+            <ListItemText primary={item.text} />
+          </ListItemButton>
+        ))}
+      </List>
+    </Box>
+  );
+};
+
```
Comment (itsvick) [inline src/components/SideBar.tsx:58]: @suvarnakale test sidebar for responsive

File: src/pages/QuestionSetEditor.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import React, { useEffect, useRef } from "react";
+import { questionSetEditorConfig } from "../components/data";
```
Comment (itsvick) [inline src/pages/QuestionSetEditor.js:2]: @suvarnakale make this file ts

File: src/pages/workspace/content/allContents/index.tsx [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+          <SearchBox
+            placeholder="Search by title..."
+            onSearch={handleSearch}
+            onFilterChange={handleFilterChange}
+            onSortChange={handleSortChange}
+          />
+        </Box>
+
+        <Table>
+          <TableHead>
```
Comment (itsvick) [inline src/pages/workspace/content/allContents/index.tsx:159]: @suvarnakale for table use same library used in admin portal

**Review by itsvick — CHANGES_REQUESTED** [review]

File: src/pages/workspace/content/allContents/index.tsx (hunk 11 above)
Comment (suvarnakale) [reply src/pages/workspace/content/allContents/index.tsx]: ok

File: src/components/CourseCard.tsx (hunk 2 above)
Comment (suvarnakale) [reply src/components/CourseCard.tsx:46]: ok

File: src/components/CourseCard.tsx (hunk 4 above)
Comment (suvarnakale) [reply src/components/CourseCard.tsx:64]: ok

File: src/components/CourseCard.tsx (hunk 5 above)
Comment (suvarnakale) [reply src/components/CourseCard.tsx:70]: ok

File: src/components/CourseCard.tsx (hunk 6 above)
Comment (suvarnakale) [reply src/components/CourseCard.tsx:76]: ok

File: src/components/CourseCard.tsx (hunk 7 above)
Comment (suvarnakale) [reply src/components/CourseCard.tsx:82]: ok

File: src/components/SearchBox.tsx (hunk 8 above)
Comment (suvarnakale) [reply src/components/SearchBox.tsx]: its dumy.. we will have state, district board filter here.

File: src/components/SideBar.tsx (hunk 9 above)
Comment (suvarnakale) [reply src/components/SideBar.tsx:58]: ok

File: src/pages/workspace/content/allContents/index.tsx (hunk 11 above)
Comment (suvarnakale) [reply src/pages/workspace/content/allContents/index.tsx]: use KA table library but our table has complex data and icons, KA is not supporting it

File: src/pages/QuestionSetEditor.js (hunk 10 above)
Comment (suvarnakale) [reply src/pages/QuestionSetEditor.js:2]: this file is replaced. with Editor.tsx page

File: src/components/SideBar.tsx (hunk 9 above)
Comment (suvarnakale) [reply src/components/SideBar.tsx:58]: will assign this to Shreyas.

File: src/components/CourseCard.tsx (hunk 3 above)
Comment (suvarnakale) [reply src/components/CourseCard.tsx:55]: done

**Pull request merged**